client/src/productDetails/components/AddToCart.jsx

```diff
@@ -214,7 +239,7 @@ const AddToCart = ({
       </SizeAndQuantity>
       <AddBagAndFavorite>
         <AddtoBagBtn onClick={addToCart} type="submit">ADD TO BAG +</AddtoBagBtn>
-        <FavoriteBtn onClick={saveFavorite} type="button">★</FavoriteBtn>
+        {favorite}
```

Comment:
This is totally fine, but you could also use conditional rendering here as another option